### ISSUE TITLE
`DecodeJson` crashes

### ISSUE THREAD

**Issue by curiousleo** [jart/cosmopolitan#475]
I used [Nicolas Seriot's JSONTestSuite](https://github.com/nst/JSONTestSuite) to test the new JSON parser introduced for #247.

Some of them crash the redbean process:

- `i_structure_500_nested_arrays.json`:  stack overflow
- `n_object_repeated_null_null.json`:  segv_maperr
- `n_structure_100000_opening_arrays.json`:  stack overflow
- `n_structure_open_array_object.json`:  stack overflow

I think this is problematic: if `DecodeJson` is used to parse user-supplied JSON, then users could easily crash the redbean process.

Full repro script to run the whole test suite, including the files that crash the parser:

```bash
#!/usr/bin/env bash
set -Eeuo pipefail

tmp="$(mktemp -d)"
git clone https://github.com/nst/JSONTestSuite.git "${tmp}"
redbean="${tmp}/redbean.com"
curl -sSo "${redbean}" https://redbean.dev/redbean-2.0.11.com
chmod +x "${redbean}"

dir="${tmp}/test_parsing"
for f in ${dir}/*.json; do
    "${redbean}" -i -e "DecodeJson(unix.read(unix.open('${f}', unix.O_RDONLY))); unix.exit(0)" >/dev/null || echo "${f}"
done
```

[redbean-json-log.txt](https://github.com/jart/cosmopolitan/files/9084523/redbean-json-log.txt)

**Comment by ahgamut** [comment]
@curiousleo thanks for testing! I had used the test cases for JSON from https://www.json.org/JSON_checker/ but the repo you've linked is more comprehensive.

Could you try this test suite again by building redbean with my branch for #476? With a depth limit for JSON, redbean should now error out with `maximum depth exceeded` instead of segfaulting.

**Comment by curiousleo** [comment]
Amazing. I tested against 6ee1898 (since this is the latest commit on `main` that touched JSON decoding) and could not get it to crash with Nicolas Seriot's examples. I can see that you've included them into unit tests, which I think is a great idea.

So I consider the original issue resolved.

The one comment I have about _how_ this was resolved is the handling of "lonely null" and "lonely false".

The behaviour is clearly documented in https://github.com/jart/cosmopolitan/blob/6ee18986e4047499e26102c788f6e70bb6dec2bc/tool/net/help.txt#L695-L696

And the two tests check that it is in fact the case:

https://github.com/jart/cosmopolitan/blob/6ee18986e4047499e26102c788f6e70bb6dec2bc/test/tool/net/samples/JSONTestSuite-pass.lua#L52-L53

https://github.com/jart/cosmopolitan/blob/6ee18986e4047499e26102c788f6e70bb6dec2bc/test/tool/net/samples/JSONTestSuite-pass.lua#L61-L62

However, the original tests are meant to check if parsers successfully parse those two examples (that's why the original files are prefixed with `y_`). So I thought that perhaps there was an issue in the test with "falsey" handling and tried to "fix" it:

```diff
-assert(not DecodeJson([[ false ]])) 
+assert(false == DecodeJson([[ false ]]))
```

That test failed, and I eventually found the documentation that says this is the expected behaviour.

To avoid this confusion, my suggestion would be to document very clearly in the tests the instances where `DecodeJson` deviates from the expectations of the imported JSONTestSuite.

**Comment by jart** [comment]
I agree. I audited every single JSON test and annotated each one that might not be in line with expectations. See https://github.com/jart/cosmopolitan/commit/3f3e7e92d795907a58b990bee531845e639a8f8e I managed to find a way to make it crash, and then fixed it. Our JSON parser is now even stricter. It's better documented. `VisualizeControlCodes` now offers protection against Trojan Source. Many peculiar behaviors from v8 have now been made consistent with ljson, such as encoding Infinity as null.
